### PR TITLE
[OneDrive] Add support for field mime-type

### PR DESCRIPTION
**Pull request by praveen-elastic**
## Closes https://github.com/elastic/connectors-py/issues/2722

This PR adds a new field `mime-type` to be indexed in onedrive connector.

#### Pre-Review Checklist
- [x] this PR does NOT contain credentials of any kind, such as API keys or username/passwords (double check `config.yml.example`)
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] if there is no GH issue, please create it. Each PR should have a link to an issue
- [x] this PR has a thorough description
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)
- [ ] Considered corresponding documentation changes
- [ ] Contributed any configuration settings changes to the configuration reference
- [ ] if you added or changed Rich Configurable Fields for a Native Connector, you made a corresponding PR in [Kibana](https://github.com/elastic/kibana/blob/main/packages/kbn-search-connectors/types/native_connectors.ts)

#### Changes Requiring Extra Attention

- [ ] Security-related changes (encryption, TLS, SSRF, etc)
- [ ] New external service dependencies added.

## Release Note

Following PR includes support for a new field `mime-type` in OneDrive connector.

### CODE REVIEW

**Comment by praveen-elastic** [comment]
Buildkite test this

File: connectors/sources/onedrive.py [hunk 1]
```diff
@@ -581,6 +581,7 @@ def prepare_doc(self, file):
             "created_at": file.get("createdDateTime"),
             "size": file.get("size"),
             "url": file.get("webUrl"),
+            "mime_type": file.get("file", {}).get("mimeType"),
```
Comment (navarone-feekery) [inline connectors/sources/onedrive.py:584]: This can throw an error if `file` is `None`:

```
>>> test = { "file": None }
>>> test.get("file", {})
>>> test.get("file", {}).get("mimeType")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'NoneType' object has no attribute 'get'
```

Can you change it to cover this edge case?

Comment (navarone-feekery) [reply connectors/sources/onedrive.py:584]: Also could you add a test to cover if `file` is `None`?

**Review by navarone-feekery — APPROVED** [review]
🚢
